blocks/algorithms/__init__.py

```diff
@@ -164,17 +167,20 @@ class GradientDescent(DifferentiableCostMinimizer):
     """
     def __init__(self, step_rule=None, gradients=None, **kwargs):
         super(GradientDescent, self).__init__(**kwargs)
-        self.gradients = (
-            gradients if gradients
-            else dict(
-                zip(self.params, tensor.grad(self.cost, self.params))))
+        self.gradients = gradients
+        if not self.gradients:
+            logger.info("Taking the cost gradient")
```

Comment:
I would make this `logger.debug` and not `info`, it's pretty detailed information.